### DISCUSSION TITLE
SLA Logical Date is incorrect, SLA Miss Table missing entries for DAGs with catchup=true

### DISCUSSION THREAD

**Discussion by CYarros10** [apache/airflow#32728] [General]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### What happened

Airflow 2.5.1

I created a DAG with:

```
default_args = {
    "owner": "me",
    "depends_on_past": False,
    "email": [""],
    "email_on_failure": False,
    "email_on_retry": False,
    "retries": 1,
    "retry_delay": timedelta(minutes=2),
    "start_date": datetime(2023, 7, 1),
    "sla": timedelta(minutes=60),
}
with models.DAG(
    f"the_new_reporting_dag_{VERSION}",
    description=f"Export metadata to BigQuery location and create reporting table.",
    schedule_interval="0 0 * * *", # daily
    tags=tags,
    default_args=default_args,
    user_defined_macros=user_defined_macros,
    is_paused_upon_creation=True,
    catchup=True,
    max_active_runs=1,
    dagrun_timeout=timedelta(minutes=30),
) as dag:
```

and ran it on 7/7/2023.

This started dag runs with logical dates of 7/1/2023, 7/2/2023, 7/3/2023, 7/4/2023, 7/5/2023, 7/6/2023.

<img width="1479" alt="Screenshot 2023-07-07 at 1 11 26 PM" src="https://github.com/apache/airflow/assets/6383133/3b428f9a-04a8-4e31-bf48-412e12f0685c">

The SLA Miss table SHOULD have these logical dates. Yet it does not. For whatever reason, SLA Miss entries were for  2023-07-03 00:00:00, 2023-07-04 00:00:00, 2023-07-05 00:00:00, 2023-07-06 00:00:00, 2023-07-07 00:00:00, all with SLA timestamp of 2023-07-07, 16:55:54

<img width="1694" alt="Screenshot 2023-07-07 at 12 57 04 PM" src="https://github.com/apache/airflow/assets/6383133/502bf9e2-b512-4219-b3b8-8ae85ef34385">

Why? SLA Miss table seems to have incorrectly inserted the wrong logical dates.

### What you think should happen instead

SLA Miss entries should have occurred for the correct logical dates of:

2023-07-01 00:00:00
2023-07-02 00:00:00
2023-07-03 00:00:00
2023-07-04 00:00:00
2023-07-05 00:00:00
2023-07-06 00:00:00

Because they all missed their 60 minute SLA (since they were all ran on 2023-07-07)

OR

We need to rename Logical Date to Data Interval End Date in the SLA Miss table.

### How to reproduce

1. Create a DAG with an SLA of 60 minutes, start date of 6 days ago, and catchup = true.

### Operating System

cloud composer on GCP

### Deployment

Google Cloud Composer

### Deployment details

Airflow 2.5.1, Cloud Composer 2.3.3

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by CYarros10** [comment]
I'm beginning to think that SLA Miss Logical Date != DAG Run Logical Date. 

SLA Miss Logical Date == DAG Run Data Interval End Date. 

But if that is true, the SLA Miss Table is still not entering the first DAG Run Data Interval End Date of 2023-07-02 00:00:00

**Comment by potiuk** [comment]
FYI. SLA feature is broken and we know it. There is an effort to redefine how it works https://cwiki.apache.org/confluence/display/AIRFLOW/%5BWIP%5D+AIP-57+SLA+as+a+DAG-Level+Feature where you can find what's broken for it - and it's not very likely there will be any fix for what you observe.

**Comment by nathadfield** [comment]
@CYarros10 Given what we know about the nature of the SLA feature and the current efforts to try and change how it works, I don't think there's much that can be done on this issue so I am tempted to close it.  I would encourage you however to speak with those involved on the AIP above to see if your needs can be met in the future.

**Comment by potiuk** [comment]
Yep. I will convert it into dicsusion. The new SLA AIP is being voted as we speak
